clash-lib/src/Clash/Primitives/DSL.hs

```diff
@@ -840,7 +840,7 @@ compInBlock compName inPorts0 outPorts0 =
 
 -- | Convert a 'LitHDL' to a 'TExpr'
 --
--- __N.B.__: Clash 1.6 changed 'instDecl'\'s type signature. Where it would
+-- __N.B.__: Clash 1.8 changed 'instDecl'\'s type signature. Where it would
 --           previously accept 'LitHDL' in its generics/parameters argument, it
 --           now accepts a 'TExpr'. This function is mostly there to ease this
 --           transition.
```

Comment:
Example of what that looks like rendered:
https://hackage.haskell.org/package/base-4.18.0.0/docs/Data-Bits.html#v:bitSizeMaybe

Reply:
Hmmm... I'm not incredibly impressed with `@since`. This bit sounds nice:
> `@since` annotation can be used to convey information about when the function was introduced or when it has changed in the way significant to the user.

Right, so two purposes: when it was introduced, or when things were significantly changed. But then it continues:

> While multiple such annotations are not an error, only the one to appear in the comment last will be used.

Well, for when a function was introduced, that's not a problem. I don't think we particularly need that functionality, though. But what about when things were significantly changed? You might end up trying to write:

```
@since 1.8 @foo@ has feature A.

@since 1.12 @foo@ additionally has feature B.
```

_But you can't write that._ By the time you're writing the bit about feature B, you'll have to convert those paragraphs to regular paragraphs or something similar. And if you don't realise it and don't check, you'll end up just with the latter paragraph because it **silently** drops the other one.¹

I'm inclined to think `@since` is mostly useful for indicating when a function was introduced, and not for the other stated purpose.

¹ Might this explain the curious phrasing "changed in _the_ way significant to the user"? It can only document _one_ way, so that will be _the_ way indeed, it's unique. 😜

Reply:
Sounds like we should solely use it for 

>  annotation can be used to convey information about when the function was introduced

then